### DISCUSSION TITLE
What are the specific meaning of two error messages below?

### DISCUSSION THREAD

**Discussion by SherlpY** [AMReX-Codes/amrex#2724] [Q&A]
I'm using OPAL(Object Oriented Particle Accelerator Library) which uses AMReX as a black box to calculate space charge field. And when I changed the computation domain, some error occurred. At some domain it works fine but at others, there would be two error messages in different unsuccessful running process(they didn't occur at the same time) which are
"amrex::Abort::0::MLMG failed!!!" and
"amrex::Abort::0::AmrMesh::MakeNewGrids: how did this happen? !!!"

Could someone tell me what's the specific meaning of these error messages as I couldn't understand them via source code?

**Comment by WeiqunZhang** [comment]
Could you give more detail about how you changed the computation domain?  What's the domain size?

The first error message means the multigrid solver has failed to converge.  Most likely the problem is not suitable for the geometric multigrid method.  For example if the AMReX cell-centered solver is given a problem with 257^3, it cannot coarsen the grid because the size is not divisible by 2.  Then the solver is likely to fail.

The error message from AmrMesh is probably also from an unconventional domain size.

By default, AMReX will check the input parameters to make sure they are compatible with the assumptions in AmrMesh.  Maybe our check fails to catch the issue, or maybe the check is disabled in OPAL.

**Reply by SherlpY** [reply]
Thank you for your reply. It's not only domain size that affects, but the number of grid points in one dimension also counts. The corresponding settings are at 1.4 and 1.5 at https://gitlab.psi.ch/OPAL/documentation/manual/blob/master/fieldsolvers.asciidoc.

I set MX=8,MY=8,MZ=4,AMR_DOMAIN_RATIO={1.0,1.0,1.0} it works.
I set MX=4,MY=4,MZ=4,AMR_DOMAIN_RATIO={1.0,1.0,1.0} it fails for MLMG failed
I set MX=4,MY=4,MZ=4,AMR_DOMAIN_RATIO={1.0,0.75,0.75} it works

Maybe other attributes count too but I haven't tested yet.

And the developer of OPAL said they didn't disable the check.
